### DISCUSSION TITLE
Upcoming changes in 5.0.

### DISCUSSION THREAD

**Discussion by lukef** [lukef/qr.flutter#168] [Announcements]
Hi All.

With the upcoming 5.0 release there will be a whole slew of new features, fixes and (more importantly) changes.

**5.0 will break compatibility with all versions <= 4.0 in a pretty big way.**

In the effort to support more configuration, the code is moving toward formalizing the styling part of the QR code from the data portion of the QR code. That means moving a whole lot of stuff around.

It's still in progress (**no firm release date yet**) but it's looking great. 

I know breaking compatibility is annoying, and Ihope that this won't be a recurring thing, but I do think this gives a really strong foundation for more cool stuff. Well .. as cool as a QR code can be 😄 

Stay tuned for more.

**Comment by lukef** [comment]
re: dates. I'm REALLY hoping to get a release out within the next 3-5 days.

**Comment by lukepighetti** [comment]
:shipit:

**Comment by jlafazia-figure** [comment]
While we wait for v5.0 would someone mind publishing v4.0.1 to pub.dev so we can get the update with qr v^3.0.0?
I have a dependency on printing AND qr_flutter and can't update printing until qr_flutter depends on qr v^3.0.0.

**Comment by ewankirk** [comment]
It really would be good to publish this to pub dev.  The dependency on QR 3.0.0 is causing me problems too.

**Comment by hagerf** [comment]
@lukef From what I can see in the repo it seems that this project has been abandoned. Issues and PRs are ignored and the last release was 2 years ago. Could you just confirm that this is the case? 
Thanks

**Comment by lukef** [comment]
In all honesty, yes. I don't really have the time or the motivation to update this repo anymore. I no longer participate in the Flutter community, nor do I work with Flutter. It's a shame, and I won't go into the reasons, but I ultimately don't have any drive to participate in the community. 

I'd be open to handing the reigns to someone else if they're interested.

I appreciate the direct question. Sorry if it's not the answer you wanted.

**Comment by hagerf** [comment]
Thanks for the quick answer, it was totally expected. And it's totally fine if you don't have the time, nor do I 😁. When choosing what libraries to use it always helps to know for certain if a repo is active or not, so thanks again for clearing the fog
